test/e2e/operator_test.go

```diff
@@ -3035,6 +3035,51 @@ func newLoadBalancerController(name types.NamespacedName, domain string) *operat
 	}
 }
 
+// TestIngressOperatorCacheIsNotGlobal validates BZ2075671: Don't include all objects in all namespaces in the
+// Ingress Operator's cache. This tests adds an Ingress Controller in another namespace that isn't in the
+// Ingress Operator's cache and ensures the Ingress Operator ignores it.
+func TestIngressOperatorCacheIsNotGlobal(t *testing.T) {
+	// Setup a namespace that will be ignored by the ingress operator.
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ignored-ns",
+		},
+	}
+	if err := kclient.Create(context.TODO(), ns); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), ns); err != nil {
+			t.Fatalf("failed to delete namespace %v: %v", ns.Name, err)
+		}
+	}()
+	// Create the new private controller in a namespace that will be ignored by the Ingress Operator.
+	icName := types.NamespacedName{
+		Namespace: ns.Name,
+		Name:      "test-cache-is-not-global",
+	}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	// Clean up our new Ingress Controller after we are done.
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Verify new Ingress Controller is ignored by ensuring its status never gets modified.
```

Comment:
Is an unmodified status the only way to check the default IC cache?

Reply:
Seems like the best way as far as I can tell. I diffed the YAML of a IC that was admitted and one that was not. In comparison, there are a couple things different, but it seems like status is this most direct way to test  The other things are default values that get added by the ingress operator (that might change in the future).